include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -138,6 +138,18 @@ enum class DeclKind { Port, Instance, Other };
 
 DeclKind getDeclarationKind(Value val);
 
+enum class ReadPortSubfieldKind { addr, en, clk, data };
+enum class WritePortSubfieldKind { addr, en, clk, data, mask };
+enum class ReadWritePortSubfieldKind {
+  addr,
+  en,
+  clk,
+  wmode,
+  rdata,
+  wdata,
+  wmask
+};
```

Comment:
How about just "ReadPortSubfield", "WritePortSubfield", and "ReadWritePortSubfield"?